bluepysnap/kgforge.py

```diff
@@ -0,0 +1,317 @@
+"""KnowledgeGraphForge wrappings for SNAP.
+
+General notes:
+    User logging in runtime not yet implemented
+    - Did some tests, and there seems to be an issue regarding the Oauth2
+      client secret which we do not have.
+    - Should be investigated further.
+    - So for now, need to copy the token for the kgforge
+
+    Getting resources created by current user not implemented as of yet
+    - Would be simple, if we had a Oauth2 login and could get the username from there
+
+    Simulation not tested yet due to there not being Simulations with SONATA config in Nexus
+    - SNAP does not support BlueConfig
+
+    Currently the functions retrieve all the instances in the KG, also the deprecated ones
+    - if this is not wanted, there is a way to exclude them in the sparql queries
+
+    KnowledgeGraphForge instance is passed to the classes' __init__ function
+    - I would prefer a global instance of it but then it would need to be changed if accessing
+      data in other 'buckets' (org/project)
+    - not sure what is the best approach for this or if this is going to be simpler in the future.
+"""
+
+
+import os
+import logging
+from cached_property import cached_property
+
+# from bluepysnap import Circuit, Simulation
+from bluepysnap import Circuit
+from bluepy.v2 import Simulation as BluepySimulation, Circuit as BluepyCircuit
```

Comment:
I am not gonna lie this is really strange having this here...

Reply:
Can it even be merged here?

Reply:
And we have a potential cyclic dependency because snap is a dep of bluepy and bluepy is imported in snap.

Reply:
> Can it even be merged here?

Not sure about the linting. Maybe if the import is lazy.

Reply:
I would avoid bluepy here at all.

Reply:
definitely ...

Reply:
I think @mgeplf this is your call here. 
To me the whole kgforge thing should be in another package to be honest. But I will follow your call.

Reply:
I have no objection to this being its own package. 
Regarding Bluepy, I was under the impression that it will not stay here.

Reply:
Yeah, we can't really have bluepy refs in here; can we make an extension point that bluepy can hook to make this work?  I'd prefer it not be a monkey patch situation, for cleanliness sake.